language/move-prover/bytecode/src/borrow_analysis.rs

```diff
@@ -420,6 +421,35 @@ impl<'a> BorrowAnalysis<'a> {
             })
             .collect();
     }
+
+    fn get_vec_bor_mut_mid(&self) -> Option<move_model::model::ModuleId> {
```

Comment:
Given that this linear search will happen with every single function call we visit, we can't do it like this.

Please precompute the `QualifiedId<FunId>` of the borrow_vec_mut function and store it in the state of the borrow analysis. Then the search would still happen for each function we analysis, so better even, store it as an extension in the global env (GlobalEnv::set_extension). You can do that by adding fn initialize to the FunctionTargetProcessor trait implementation.

Reply:
Nice catch. Meant to memoize this but completely forgot. Adding to globalenv sounds even better. Will do that.